.github/workflows/gap.yml

```
@@ -0,0 +1,75 @@
+name: CI with GAP
+
+on:
+  push:
+    branches:
+      - 'master'
+      - 'release-*'
+    tags: '*'
+  pull_request:
+
+concurrency:
+  # group by workflow and ref; the last slightly strange component ensures that for pull
+  # requests, we limit to 1 concurrent job, but for the default repository branch we don't
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.ref_name != github.event.repository.default_branch || github.run_number }}
+  # Cancel intermediate builds, but only if it is a pull request build.
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
+
+jobs:
+  test:
+    name: Julia ${{ matrix.julia-version }} - GAP ${{ matrix.gap-version }} - ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 20
+    continue-on-error: ${{ matrix.julia-version == 'nightly' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        gap-version:
+          - 'master'
+          - 'stable-4.13'
+          - 'v4.13.0'       # latest release
+        julia-version:
+          - '1.6'
+          - '1.10'
+          - 'nightly'
+        julia-arch:
+          - x64
+        os:
+          - ubuntu-latest
+          - macOS-latest
+        exclude:
+          # Remove a few macOS jobs (the number we can run in parallel is limited)
+          - julia-version: '1.6'
+            os: macOS-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: "Set up Julia"
+        uses: julia-actions/setup-julia@v2
+        with:
+          version: ${{ matrix.julia-version }}
+          arch: ${{ matrix.julia-arch }}
+      - name: Checkout GAP
+        uses: actions/checkout@v4
+        with:
+          repository: 'gap-system/gap'
+          ref: ${{ matrix.gap-version }}
+          path: 'GAPROOT'
+      - name: "Install dependencies (for macOS)"
+        if: runner.os == 'macOS'
+        run: |
+          brew install autoconf gmp zlib pkg-config
```

```suggestion
          brew install autoconf zlib # gmp pkg-config
```
These are already installed on the runner